packages/client-engine-runtime/src/transaction-manager/transaction-manager.test.ts

```diff
@@ -141,9 +141,47 @@ test('getTransaction works while being rolled back', async () => {
 
   const id = await startTransaction(transactionManager)
 
-  const rollbackPromise = transactionManager.rollbackTransaction(id)
-  expect(() => transactionManager.getTransaction({ id }, 'dummy')).toThrow('Transaction is being closed')
-  await rollbackPromise
+  const rollbackPromise1 = transactionManager.rollbackTransaction(id)
+  const rollbackPromise2 = transactionManager.rollbackTransaction(id)
+  await rollbackPromise1
+
+  await expect(rollbackPromise2).rejects.toEqual(new TransactionRolledBackError('rollback'))
+
+  expect(driverAdapter.rollbackMock).toHaveBeenCalled()
+  expect(driverAdapter.executeRawMock.mock.calls[0][0].sql).toEqual('ROLLBACK')
+  expect(driverAdapter.commitMock).not.toHaveBeenCalled()
+})
+
+test('commitTransaction during a rollback caused by a time out raises a TransactionExecutionTimeoutError', async () => {
+  const driverAdapter = new MockDriverAdapter()
+  const timeout = 200
+  const rollbackDelay = 200
+
+  driverAdapter.rollbackMock = jest.fn().mockImplementation(() => {
+    return new Promise((resolve) => {
+      setTimeout(() => {
+        resolve(undefined)
+      }, rollbackDelay)
+    })
```

Comment:
nit: you can
```ts
import timers from 'node:timers/promises`
```
and then replace this with
```suggestion
    return timers.setTimeout(rollbackDelay)
```

Reply:
updated